BioForm/profiles/middleware.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+from django.urls import reverse
+from django.shortcuts import redirect
+from django.contrib.auth import logout
+import re
+
+# URLs that are exempt from redirect such as login and register
+EXEMPT_URLS = [re.compile(settings.LOGIN_URL.lstrip('/'))]
+if hasattr(settings, 'LOGIN_EXEMPT_URLS'):
+    EXEMPT_URLS += [re.compile(url) for url in settings.LOGIN_EXEMPT_URLS]
+
+# Middleware class to redirect user according to authentication
+class LoginRequiredMiddleware:
```

Comment:
Middlewares are usually not used for this purpose, Django has permissions classes for this purpose. Middleware checks if a user is authenticated then the current user is attached with a request and if a user is not authenticated then An anonymous user is attached with a request.

Reply:
I will check out how to do login redirects in permissions class and try to do custom authentication using middleware then

Reply:
Done with login decorators now